dace/codegen/targets/cpu.py

```diff
@@ -1925,6 +1927,12 @@ def _generate_MapEntry(
             # Emit nested loops
             for i, r in enumerate(node.map.range):
                 var = map_params[i]
+                if var in map_param_types.keys():
+                    var_type = map_param_types[var]
+                else:
+                    var_type = dtypes.result_type_of(infer_expr_type(r[0], state_dfg.symbols_defined_at(node)), 
+                                                     infer_expr_type(r[1], state_dfg.symbols_defined_at(node)))
```

Comment:
Please compute this once, preferably outside the loop of node.map.range. `symbols_defined_at` can be an expensive function. Thanks!